### DISCUSSION TITLE
Possible? - Modify Security Group rules

### DISCUSSION THREAD

**Discussion by Splaxy** [cloud-custodian/cloud-custodian#7319] [General]
Hello,

I'm super new to the Cloud Custodian - I was curious if it's possible to have the custodian modify a "matching" security-group and then notify:

Goal: Find security groups allowing ingress port 22,3389 from 0.0.0.0 and then modify the allowed cidr to x.x.x.x, after modification, notify via normal methods.

Current code snippit below:

  name: open-sec-group
  resource: security-group
  comment: "Security group allows 22,3389 from 0.0.0.0/0"
  filters:
  - type: ingress
    Ports: [22,3389]
    Cidr:
      value: 0.0.0.0/0
      op: eq
      value_type: cidr
  mode:
    schedule: "rate(24 hours)"
    type: periodic
  actions:
    - type: notify
      action_desc: "Changed allowed CIDR from 0.0.0.0/0 to x.x.x.x/x"
      to: ["****"]
      transport:
        type: sqs
        queue: ******
      violation_desc: |
        *********

I had looked at the "patch" function but I'm finding it hard to find examples on usage.

Any help appreciated! Thanks!

**Comment by kapilt** [comment]
wrt to rule modifications, we can match on particular ingress/egress rules, and then delete them, we don't really do arbitrary mods to rules in custodian policies, we would recommend using something like terraform/cfn wrt to provisioning resources with custodian focused on governance policies. the patch functionality thats there was to support a slightly different use case of reverting a group back to a historical state (ie. a group as a locked resource would not be modifyable and modifications would trigger a reversion).

**Comment by lkolchin** [comment]
That'd be a great feature.
I hope this will eventuate and cover it:
https://github.com/cloud-custodian/cloud-custodian/pull/4973
